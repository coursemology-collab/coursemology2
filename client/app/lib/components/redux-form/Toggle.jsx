import React from 'react';
import PropTypes from 'prop-types';
import MaterialSwitch from 'material-ui/Switch';
import red from 'material-ui/colors/red';
import createComponent from './createComponent';
import mapError from './mapError';

const red500 = red['500'];

const errorStyle = {
  color: red500,
};

// Toggle implementation with an error displayed at the bottom.
const Toggle = ({ errorText, ...props }) => (
  <div>
    <MaterialSwitch {...props} />
    {
        errorText &&
        <div style={errorStyle}>{errorText}</div>
      }
  </div>
);

Toggle.propTypes = {
  errorText: PropTypes.string,
};

export default createComponent(
  Toggle,
  ({
    input: {
      onChange,
      value,
      ...inputProps
    },
    ...props
  }) => ({
    // Take out the required fields and send the rest of the props to mapError().
    ...mapError({ ...props, input: inputProps }),
    checked: !!value,
    onChange,
  })
);
