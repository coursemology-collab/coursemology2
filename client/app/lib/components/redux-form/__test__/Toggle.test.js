import React from 'react';
import shallowUntil from 'utils/shallowUntil';
import Toggle from '../Toggle';

describe('<Switch />', () => {
  it('renders a checked toggle', () => {
    const ToggleInput = shallowUntil(
      <Switch
        input={{
          name: 'Awesome toggle',
          value: true,
          onChange: jest.fn(),
        }}
      />,
      {
        context: { intl },
        childContextTypes: { intl: intlShape },
      },
      'Toggle'
    );

    expect(ToggleInput).toMatchSnapshot();
  });

  it('renders a toggle with an error', () => {
    const ToggleInput = shallowUntil(
      <Switch
        input={{
          name: 'Not very awesome toggle',
          value: false,
          onChange: jest.fn(),
        }}
        meta={{
          error: 'Ooops',
          touched: true,
        }}
      />,
      {
        context: { intl },
        childContextTypes: { intl: intlShape },
      },
      'Toggle'
    );

    expect(ToggleInput).toMatchSnapshot();
  });
});
