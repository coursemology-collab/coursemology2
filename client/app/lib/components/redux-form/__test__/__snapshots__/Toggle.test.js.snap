// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Switch /> renders a toggle with an error 1`] = `
<div>
  <Switch
    defaultToggled={false}
    disabled={false}
    labelPosition="left"
    name="Not very awesome toggle"
    onChange={[Function]}
    checked={false}
  />
  <div
    style={
      Object {
        "color": "#f44336",
      }
    }
  >
    Ooops
  </div>
</div>
`;

exports[`<Switch /> renders a checked toggle 1`] = `
<div>
  <Switch
    defaultToggled={false}
    disabled={false}
    labelPosition="left"
    name="Awesome toggle"
    onChange={[Function]}
    checked={true}
  />
</div>
`;
